rfc/0008-builtin-blocktypes/README.md

```diff
@@ -0,0 +1,212 @@
+- `property` keyword to define a property with a name and a type
+  - Optionally, a default value can be assigned
+- Adds new keywords for existing property types:
+  - Valuetypes: `regex`, `cellrange` / `row` / `column` / `cell`, `valuetype-assignment`
+  - Typed collections: `collection<type>`
+    - Nested collections are not supported for now
+  - IO types: `File`, `FileSystem`, `TextFile`, `Sheet`, `Table`
+
+See the following section for concrete code examples.
+
+### Syntax for current block types
+
+<details>
+<summary>Click to show</summary>
+
+```jayvee
+builtin blocktype HttpExtractor {
+    property url oftype text;
+
+    output oftype File;
+}
+
+builtin blocktype ArchiveInterpreter {
+    input oftype File;
+
+    property archiveType oftype text;
+
+    output oftype FileSystem;
+}
+
+builtin blocktype FilePicker {
+    input oftype FileSystem;
+
+    property path oftype text;
+
+    output oftype File;
+}
+
+builtin blocktype TextFileInterpreter {
+    input oftype File;
+
+    property encoding oftype text;
+    property lineBreak oftype regex;
+
+    output oftype TextFile;
+}
+
+builtin blocktype TextLineDeleter {
+    input oftype TextFile;
+
+    property lines oftype collection<integer>;
+
+    output oftype TextFile;
+}
+
+builtin blocktype TextRangeSelector {
+    input oftype TextFile;
+
+    property lineFrom oftype integer;
+    property lineTo oftype integer;
+
+    output oftype TextFile;
+}
+
+builtin blocktype CSVInterpreter {
+    input oftype TextFile;
+
+    property delimiter oftype text: ",";
+    property enclosing oftype text: "";
+    property enclosingEscape oftype text: "";
+
+    output oftype Sheet;
+}
+
+builtin blocktype CellRangeSelector {
+    input oftype Sheet;
+
+    property select oftype cellrange;
+
+    output oftype Sheet;
+}
+
+builtin blocktype CellWriter {
+    input oftype Sheet;
+
+    property write oftype text;
+    property at oftype cell;
+
+    output oftype Sheet;
+}
+
+builtin blocktype ColumnDeleter {
+    input oftype Sheet;
+
+    property delete oftype collection<column>;
+
+    output oftype Sheet;
+}
+
+builtin blocktype RowDeleter {
+    input oftype Sheet;
+
+    property delete oftype collection<row>;
+
+    output oftype Sheet;
+}
+
+builtin blocktype TableInterpreter {
+input oftype Sheet;
+
+    property header oftype boolean;
+    property columns oftype collection<valuetype-assignment>;
+
+    output oftype Table;
+}
+
+builtin blocktype SQLiteLoader {
+    input oftype Table;
+
+    property table oftype text;
+    property file oftype text;
+}
+
+builtin blocktype PostgresLoader {
+    input oftype Table;
+
+    property host oftype text;
+    property port oftype integer;
+    property username oftype text;
+    property password oftype text;
+    property database oftype text;
+    property table oftype text;
+}
+```
+</details>
+
+## Drawbacks
+
+- Adds valuetype keywords for properties, but these types cannot be assigned to table columns 
```

Comment:
Does this mean the IOTypes keywords like Table? I would consider these extra M2 concepts and not value types, then being able to use them in oftype but not for table columns is the same as using blocktypes in oftype for blocks bot nowhere else.

Reply:
Ah reading on, I assume this means types like collection, valuetype-assignment etc. I don't think it's a big problem.